ocean_lib/data_provider/data_service_provider.py

```diff
@@ -366,7 +375,10 @@ def build_endpoint(service_name, provider_uri=None, config=None):
         if api_version not in provider_uri:
             provider_uri = f'{provider_uri}/{api_version}'
 
-        return f'{provider_uri}/services/{service_name}'
+        valid_provider_info = DataServiceProvider.get_service_endpoints()
+        method, url = valid_provider_info['serviceEndpoints'][service_name]
```

Comment:
see comment above, you can remove `['serviceEndpoints']` and maybe rename `valid_provider_info` to `service_endpoints`